libjpeg-turbo/CMakeLists.txt

```diff
@@ -0,0 +1,134 @@
+    set(opt_flags "-Og ${common_flags} ${assert_flags}")
+    set(opt_args -DCMAKE_BUILD_TYPE=Release
+    -DCMAKE_C_FLAGS_RELEASE=${opt_flags}
+    -DCMAKE_CXX_FLAGS_RELEASE=${opt_flags})
+elseif(CONFIG_COMPILER_OPTIMIZATION_SIZE)
+    # We redefine the flags for Release instead.
+    set(opt_flags "-Os ${common_flags} ${assert_flags}")
+    set(opt_args -DCMAKE_BUILD_TYPE=Release
+                 -DCMAKE_C_FLAGS_RELEASE=${opt_flags}
+                 -DCMAKE_CXX_FLAGS_RELEASE=${opt_flags})
+elseif(CONFIG_COMPILER_OPTIMIZATION_PERF)   
+    set(opt_flags "-O3 ${common_flags} ${assert_flags}")
+    set(opt_args -DCMAKE_BUILD_TYPE=Release
+                 -DCMAKE_C_FLAGS_RELEASE=${opt_flags}
+                 -DCMAKE_CXX_FLAGS_RELEASE=${opt_flags})
+elseif(COMPILER_OPTIMIZATION_NONE)
+    set(opt_flags "-O0 ${common_flags} ${assert_flags}")
+    set(opt_args -DCMAKE_BUILD_TYPE=Debug
+                 -DCMAKE_C_FLAGS_DEBUG=${opt_flags}
+                 -DCMAKE_CXX_FLAGS_DEBUG=${opt_flags})
+else()
+    set(opt_flags "-O0 ${common_flags} ${assert_flags}")
+    set(opt_args -DCMAKE_BUILD_TYPE=Debug
+                 -DCMAKE_C_FLAGS_DEBUG=${opt_flags}
+                 -DCMAKE_CXX_FLAGS_DEBUG=${opt_flags})
+endif()
+
+idf_build_get_property(python PYTHON)
+
+include(ExternalProject)
+
+# Build jpeg-turbo in this directory:
+set(BINARY_DIR ${CMAKE_CURRENT_BINARY_DIR}/libjpeg-build)
+
+# The list here just to make it common with other components
+set(jpegturbo_libraries
+    libjpeg
+)
+
+set(all_libraries)
+set(all_targets)
+
+foreach(libname ${jpegturbo_libraries})
+    set(lib_path ${BINARY_DIR}/install/lib/${libname}.a)
+    list(APPEND all_libraries ${lib_path})
+    list(APPEND all_targets libjpeg_${libname})
+    add_prebuilt_library(libjpeg_${libname} ${lib_path})
+endforeach()
+
+# Add jpeg-turbo as a subproject.
+ExternalProject_Add(jpegturbo_proj
+    SOURCE_DIR ${COMPONENT_DIR}/libjpeg-turbo
+    BINARY_DIR ${BINARY_DIR}
+    BUILD_BYPRODUCTS ${all_libraries}
+    # These two options are set so that Ninja immediately outputs
+    # the subproject build to the terminal. Otherwise it looks like the
+    # build process "hangs" for too long until jpeg-turbo build is complete.
+    USES_TERMINAL_CONFIGURE TRUE
+    USES_TERMINAL_BUILD TRUE
+    # Arguments to pass to jpeg-turbo CMake invocation:
+    CMAKE_ARGS
+        -DCMAKE_C_FLAGS=${c_flags}
+        -DCMAKE_CXX_FLAGS=${cxx_flags}
+        ${opt_args}
+        -DCMAKE_INSTALL_PREFIX=${BINARY_DIR}/install
+        -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TOOLCHAIN_FILE}
+        -DWITH_TURBOJPEG=NO
+        -DWITH_SIMD=NO
+        -DCMAKE_SYSTEM_PROCESSOR=esp32 # Without this parameter the lib make an error
+        -DWITH_ARITH_DEC=YES
+        -DWITH_ARITH_ENC=YES
+        -DWITH_JPEG8=YES
+        -DWITH_JPEG7=YES
+        -DENABLE_SHARED=NO
+        -DENABLE_STATIC=YES        
+)
+
+add_dependencies(${COMPONENT_LIB} jpegturbo_proj)
+set(CMAKE_POSITION_INDEPENDENT_CODE ON)
```

Comment:
Should this not be passed to `CMAKE_ARGS` argument of `ExternalProject_Add` above?